core/node/node_framework/src/implementations/layers/pk_signing_eth_client.rs

```diff
@@ -86,15 +92,40 @@ impl WiringLayer for PKSigningEthClientLayer {
                 private_key.clone(),
                 self.contracts_config.diamond_proxy_addr,
                 gas_adjuster_config.default_priority_fee_per_gas,
-                self.sl_chain_id,
+                l1_chain_id,
                 query_client,
             );
             BoundEthInterfaceForBlobsResource(Box::new(signing_client_for_blobs))
         });
 
+        let signing_client_for_gateway = if let (Some(client), Some(gateway_contracts)) = (
+            &input.gateway_client,
+            self.gateway_contracts_config.as_ref(),
+        ) {
+            if gateway_contracts.gateway_chain_id.0 != 0u64 {
+                let private_key = self.wallets.operator.private_key();
```

Comment:
Dumb question: Is it intentional that the `operator` wallet is used here, not the new `gateway` wallet?

Reply:
I looked through the code, the `gateway` wallet is not used at all atm. Since we do not need the operator addresses to differ and it is already how the server is implemented, I'll delete it from the wallet